blog/_posts/2022-03-15-pulumi.md

```diff
@@ -0,0 +1,275 @@
+        }] 
+      }
+    }
+  }
+});
+exports.name = deployment.metadata.name;
+```
+
+And because this is a real programming language, it's easy to factor out the
+common constants.
+
+```javascript
+"use strict";
+const k8s = require("@pulumi/kubernetes");
+
+const appLabels = { app: "pulumi-helloworld" };
+const deployment = new k8s.apps.v1.Deployment("pulumi-helloworld", {
+  spec: {
+    selector: { 
+      matchLabels: appLabels 
+    },
+    replicas: 1,
+    template: {
+      metadata: { labels: appLabels },
+      spec: { 
+        containers: [{ 
+          name: "pulumi-helloworld", 
+          image: "pulumi-helloworld-image" 
+        }] 
+      }
+    }
+  }
+});
+exports.name = deployment.metadata.name;
+```
+
+## Setting up a Fast Local Dev Loop
+
+I've built a lot of developer tools! And I started to notice how every tool
+would gradually expand in scope to add the same set of features to make them
+iterating on them easy:
+
+- File watches and dependency tracking for auto-builds.
+
+- Terminal & UI dashboards for real-time status.
+
+- Diagnostic CLIs for inspecting when things went wrong.
+
+We built Tilt to be a dev environment that integrates with any build/deploy tool.
+That's why Tilt has a pluggable system for deploying to Kubernetes, such that we
+can layer on our own image builds and live updates for local development.
+
+Let's take a look at how it works!
+
+First, we modify our Pulumi script to take an image as an argument:
+
+```javascript
+"use strict";
+const pulumi = require("@pulumi/pulumi");
+const k8s = require("@pulumi/kubernetes");
+
+let config = new pulumi.Config();
+let image = config.require("image");
+
+const appLabels = { app: "pulumi-helloworld" };
+const deployment = new k8s.apps.v1.Deployment("pulumi-helloworld", {
+    spec: {
+        selector: { matchLabels: appLabels },
+        replicas: 1,
+        template: {
+            metadata: { labels: appLabels },
+            spec: { containers: [{ name: "pulumi-helloworld", image: image }] }
+        }
+    }
+});
+exports.name = deployment.metadata.name;
+```
+
+Now, we use Tilt to build the image, live update the running Pod, and connect a port-forward
+from `localhost:8000` to the container deployed with Pulumi.
+
+Tilt's `pulumi` extension invokes `pulumi up` to deploy to the cluster.
+
+```python
+load('ext://pulumi', 'pulumi_resource')
+
+docker_build(
+  'pulumi-helloworld-image',
+  './helloworld',
+  live_update=[
+    sync('./helloworld', '/app'),
+  ])
+
+pulumi_resource(
+  'helloworld',
+  stack='dev',
+  dir='./',
+  deps=['./index.js'],
+  image_deps=['pulumi-helloworld-image'],
+  image_configs=['image'],
+  labels=['helloworld'],
+  port_forwards=['8000:8000'])
+```
+
+The `pulumi_resource` function has arguments to define:
+
+- The name of your resource in the Tilt UI.
+
+- The name of the stack in Pulumi (this is usually implicit when you use the Pulumi CLI).
+
+- What file changes should trigger a re-deploy.
```

Comment:
Might be worth clarifying this is "Pulumi config files" or something, i.e. that files for the image are already taken care of [and will trigger a re-deploy whenever re-built]